### ISSUE TITLE
Add tests for repository._transform

### ISSUE THREAD

**Issue by marco-c** [mozilla/bugbug#384]
Also change the function name, since we're there.

**Comment by rajibmitra** [comment]
I would like to work on this , could you please assign this me @marco-c

**Comment by marco-c** [comment]
@rajibmitra feel free to work on it! Issues are assigned automatically when there is a PR to fix them.

**Comment by Iantonga** [comment]
hello I would like to work on this issue, could you give me some pointers on where to get started? Thanks!

**Comment by marco-c** [comment]
@lantonga we need to add tests for the _transform function of the repository module.

**Comment by Nathan-Wisner** [comment]
Can someone more intimate with the _transform function either add comments or give us pseudocode / an oversimplification so we can start understanding it?

**Comment by marco-c** [comment]
The function is analyzing a commit and gathering a few metrics from it. What parts of the function are not clear? If you have specific doubts I can help clarifying them.

**Comment by alihaiderrizvi** [comment]
I'm having trouble installing `rs_parsepatch`. I installed all dependencies from `requirements.txt` and `test-requirements.txt` except `rs_parsepatch`. This is the error I'm getting:

![error1](https://user-images.githubusercontent.com/45711004/111027305-52aee180-83a4-11eb-81ae-be1756a25684.png)
![error2](https://user-images.githubusercontent.com/45711004/111027323-68bca200-83a4-11eb-96e2-b8ed01be7796.png)
![image](https://user-images.githubusercontent.com/45711004/111027340-838f1680-83a4-11eb-9f44-f6b840ebea5d.png)

**Comment by marco-c** [comment]
@alihaiderrizvi from the error, it looks like you need to install Visual Studio.

**Comment by alihaiderrizvi** [comment]
I installed Visual Studio, and now I'm getting the following error:
![error1](https://user-images.githubusercontent.com/45711004/111079765-a579bd80-84b8-11eb-8c31-33454e823c1e.png)
![error2](https://user-images.githubusercontent.com/45711004/111079770-aca0cb80-84b8-11eb-8e86-7ee3fb182c36.png)

**Comment by marco-c** [comment]
Try setting Rust nightly as your default

**Comment by alihaiderrizvi** [comment]
That worked! `rs_parsepatch` is installed.
However, when I ran pytest on `tests/test_repository.py`, many tests return the following error:
`Exception: rust-code-analysis is required for code analysis`
![error](https://user-images.githubusercontent.com/45711004/111083864-d532c080-84cc-11eb-8eb5-f9d4602464e4.png)
Am I missing a dependency?

**Comment by marco-c** [comment]
Yes, you will need to install [rust-code-analysis](https://github.com/mozilla/rust-code-analysis). You can install it by using Cargo: `cargo install rust-code-analysis-web`.

**Comment by alihaiderrizvi** [comment]
I think I've messed up somewhere while setting up the project because the errors just don't stop 🤦‍♂️
![error1](https://user-images.githubusercontent.com/45711004/111207497-efc97000-8586-11eb-9ae1-a9c9834c2235.png)
The `TypeError` is coming from the following function:
![error2](https://user-images.githubusercontent.com/45711004/111207661-22736880-8587-11eb-99c8-f11b73010e17.png)
Apologies for "bugging" too much, this is my first shot at OS contribution.

**Comment by marco-c** [comment]
@alihaiderrizvi I'd suggest using Linux or a Linux VM, you'll have a much easier life.
